Sources/Security.php

```diff
@@ -1342,4 +1344,134 @@ function frameOptionsHeader($override = null)
 	header('x-content-type-options: nosniff');
 }
 
+/**
+ * This sets the Access-Control-Allow-Origin header.
+ *
+ * @since 2.1
+ */
+function corsPolicyHeader($set_header = true)
+{
+	global $boardurl, $modSettings, $context;
+
+	if (empty($modSettings['allow_cors']))
+		return;
+
+	// We want weak security.
+	if (!empty($modSettings['cors_domains']) && $modSettings['cors_domains'] === '*')
+		$context['cors_domain'] = '*';
+
+	// If subdomain-independent cookies are on, allow sub domains. This will have issues if the forum is at forum.domain.tld and the origin we are at is another.sub.domain.tld.
+	if (empty($context['cors_domain']) && !empty($modSettings['globalCookies']) && !empty($_SERVER['HTTP_ORIGIN']) && filter_var($_SERVER['HTTP_ORIGIN'], FILTER_VALIDATE_URL) !== false)
+	{
+		$boardurl_base = implode('.', array_slice(explode('.', parse_url($boardurl, PHP_URL_HOST)), 1));
+
+		// If we find www, pop one more off.
+		if ('www' === array_slice(explode('.', parse_url($boardurl, PHP_URL_HOST)), 0, 1))
```

Comment:
Is this if case needed?
The same code is executed before the if case and inside it.